m4/check_gcc_builtin.m4

```diff
@@ -0,0 +1,23 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2024      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_GCC_BUILTIN], [
+  result="no"
+
+  AC_MSG_CHECKING([if $1 is available])
+  AC_LINK_IFELSE([AC_LANG_PROGRAM([], [
+    m4_case([$1],
+      [__builtin_ffs], [$1(0)],
```

Comment:
why not just take the arguments as a second argument to the macro and skip the whole case statement?   Also, it looks like __builtin_expect is never actually used.

Reply:
__builtin_expect is used to implement OFI_LIKELY/UNLIKELY in case GCC is used, but there is an alternative (no-op) in case that is not available.